server/src/internalClusterTest/java/org/opensearch/remotemigration/RemoteStoreMigrationAllocationIT.java

```diff
@@ -0,0 +1,929 @@
+            .indices()
+            .prepareUpdateSettings(TEST_INDEX)
+            .setSettings(
+                Settings.builder()
+                    .put("index.routing.allocation.include._name", remoteNodeName)
+                    .put("index.routing.allocation.exclude._name", allNodesExcept(remoteNodeName))
+            )
+            .execute()
+            .actionGet();
+
+        ensureGreen(TEST_INDEX);
+
+        logger.info(" --> verify allocation of primary shard");
+        RoutingTable routingTable = client.admin().cluster().prepareState().execute().actionGet().getState().getRoutingTable();
+        ShardRouting primaryShardRouting = routingTable.index(TEST_INDEX).shard(0).primaryShard();
+        assertAllocation(primaryShardRouting, remoteNode);
+    }
+
+    // tests for replica shard copy allocation with REMOTE_STORE direction
+
+    public void testDontAllocateNewReplicaShardOnRemoteNodeIfPrimaryShardOnNonRemoteNodeForRemoteStoreDirection() throws Exception {
+        logger.info(" --> initialize cluster");
+        initializeCluster();
+
+        logger.info(" --> set mixed cluster compatibility mode");
+        setClusterMode(MIXED_MODE);
+
+        logger.info(" --> add remote and non-remote nodes");
+        addRemote = false;
+        String nonRemoteNodeName = internalCluster().startNode();
+        addRemote = true;
+        String remoteNodeName = internalCluster().startNode();
+        internalCluster().validateClusterFormed();
+        DiscoveryNode nonRemoteNode = assertNodeInCluster(nonRemoteNodeName);
+        DiscoveryNode remoteNode = assertNodeInCluster(remoteNodeName);
+        assertFalse(nonRemoteNode.isRemoteStoreNode());
+        assertTrue(remoteNode.isRemoteStoreNode());
+
+        logger.info(" --> allocate primary shard on non-remote node");
+        client.admin()
+            .indices()
+            .prepareCreate(TEST_INDEX)
+            .setSettings(
+                getRemoteStoreEnabledIndexSettingsBuilder(false).put("index.number_of_shards", 1)
+                    .put("index.number_of_replicas", 1)
+                    .put("index.routing.allocation.include._name", nonRemoteNodeName)
+                    .put("index.routing.allocation.exclude._name", allNodesExcept(nonRemoteNodeName))
+            )
+            .setWaitForActiveShards(ActiveShardCount.ONE)
+            .execute()
+            .actionGet();
+
+        logger.info(" --> verify allocation of primary shard");
+        RoutingTable routingTable = client.admin().cluster().prepareState().execute().actionGet().getState().getRoutingTable();
+        ShardRouting primaryShardRouting = routingTable.index(TEST_INDEX).shard(0).primaryShard();
+        assertAllocation(primaryShardRouting, nonRemoteNode);
+
+        logger.info(" --> verify non-allocation of replica shard");
+        ShardRouting replicaShardRouting = routingTable.index(TEST_INDEX).shard(0).replicaShards().get(0);
+        assertNonAllocation(replicaShardRouting);
```

Comment:
Should we include the remote store node and make sure the decision is no from the Allocator?